lib/led/rgb.js

```diff
@@ -81,11 +222,23 @@ var RGB = function(opts) {
     values: {
       get: function() {
         return RGB.colors.reduce(function(current, color) {
-          return (current[color] = this[color].value, current);
-        }.bind(this), {});
+          return (current[color] = state[color], current);
+        }, {});
+      }
+    },
+    writeColor: {
```

Comment:
I grok the motivation for splitting out this method's functionality from `write`, but I'd like to suggest renaming `writeColor` to `update` because I think it's nicer to explain as:


- `update`: updates the internal state and then calls `write`
- `write`: writes the raw value to the component

This is my last bit of change-centric feedback, after that it's just reality checks with hardware.